packages/shared-ux/page/kibana_no_data/impl/src/services.tsx

```diff
@@ -15,29 +15,30 @@ import {
 import { NoDataCardProvider, NoDataCardKibanaProvider } from '@kbn/shared-ux-card-no-data';
 
 import {
+  Services,
   KibanaNoDataPageServices,
   KibanaNoDataPageKibanaDependencies,
 } from '@kbn/shared-ux-page-kibana-no-data-types';
 
-import { LegacyServicesProvider, getLegacyServices } from './legacy_services';
-
-const KibanaNoDataPageContext = React.createContext<KibanaNoDataPageServices | null>(null);
+const KibanaNoDataPageContext = React.createContext<Services | null>(null);
 
 /**
  * A Context Provider that provides services to the component.
  */
 export const KibanaNoDataPageProvider: FC<KibanaNoDataPageServices> = ({
   children,
   ...services
-}) => (
-  <KibanaNoDataPageContext.Provider value={services}>
-    <NoDataViewsPromptProvider {...services}>
-      <NoDataCardProvider {...services}>
-        <LegacyServicesProvider {...getLegacyServices(services)}>{children}</LegacyServicesProvider>
-      </NoDataCardProvider>
-    </NoDataViewsPromptProvider>
-  </KibanaNoDataPageContext.Provider>
-);
+}) => {
+  const { hasESData, hasUserDataView } = services;
+
+  return (
+    <KibanaNoDataPageContext.Provider value={{ hasESData, hasUserDataView }}>
```

Comment:
why do we need to explicitly destructure `services` here?

Reply:
`services` could include a lot of other things, as TS types do not prevent passing objects with more than is necessary.  By destructuring first, we exclude anything extra that was passed, and keep the context value from being unintentionally bloated.